ert_shared/ensemble_evaluator/evaluator.py

```diff
@@ -93,30 +94,39 @@ def create_snapshot_msg(ee_id, snapshot, event_index):
 
     async def handle_client(self, websocket, path):
         logger.debug(f"Client {websocket.remote_address} connected.")
-        self._clients.add(websocket)
-
-        message = self.create_snapshot_msg(
-            self._ee_id, self._snapshot, self.event_index()
-        )
-        await websocket.send(message)
 
-        async for message in websocket:
-            client_event = from_json(message)
-            if client_event["type"] == identifiers.EVTYPE_EE_TERMINATE_REQUEST:
-                logger.debug(f"Client {websocket.remote_address} asked to terminate.")
-                self._stop()
-        logger.debug(f"Client {websocket.remote_address} disconnected.")
-        self._clients.remove(websocket)
+        self._clients.add(websocket)
+        try:
+            message = self.create_snapshot_msg(
+                self._ee_id, self._snapshot, self.event_index()
+            )
+            await websocket.send(message)
+
+            async for message in websocket:
+                client_event = from_json(message)
+                if client_event["type"] == identifiers.EVTYPE_EE_TERMINATE_REQUEST:
+                    logger.debug(
+                        f"Client {websocket.remote_address} asked to terminate."
+                    )
+                    self._stop()
+            logger.debug(f"Client {websocket.remote_address} disconnected.")
+        finally:
+            self._clients.remove(websocket)
```

Comment:
This is a context manager, and it should be implemented via `__enter__` and `__exit__`